### DISCUSSION TITLE
is it possible to just use the route builder?

### DISCUSSION THREAD

**Discussion by spyhunter99** [apache/camel-karavan#828] [Q&A] [answered]
assume i already have a camel based application server in my infrastructure with my own rest api's attached to it, is it possible to use the karavan UI to assist with building routes and then send it over to my rest api?

also how does the UI discover which camel components are available? I.e. i have a bunch of custom ones, does that need to be present at karavan build time or is it runtime discoverable?

**Comment by guoxunbo** [comment, marked as answer]
Yes. You can use the web designer and export the yaml/code to your owner camel application server.such as quarkus spring or ejb.
Or you can use the source code to dev a interface to upload the yaml/code to your server already. I just do it use this function.

**Comment by davsclaus** [comment]
custom components is not dynamic loaded into Karavan. It has UI support for the ones that comes OOTB from Camel via the camel-catalog JAR.
